internal/subscriber/retry/delayconsumer_test.go

```diff
@@ -81,7 +95,162 @@ func TestDelayConsumer_Run(t *testing.T) {
+	ctx, ctrl, subCtx, cancel, store, handler, consumer, cache := setupDC(t)
+	subs := getDummySubscriptionModel()
+	subscriberID := "subscriber-id"
+	errCh := make(chan error)
+	dc, err := NewDelayConsumer(subCtx, subscriberID, "t1", subs, store, handler, cache, errCh)
+	producer := buildMockProducer(ctx, ctrl)
+	store.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
+	assert.NotNil(t, dc)
+	assert.Nil(t, err)
+	assert.NotNil(t, dc.LogFields())
+	msgs := make([]messagebroker.ReceivedMessage, 0)
+	msg1 := getDummyBrokerMessage("msg-1", "msg-id-1", time.Now().Add(time.Nanosecond*1000))
+	msg2 := getDummyBrokerMessage("msg-2", "msg-id-2", time.Now().Add(time.Second*2))
+	msgs = append(msgs, msg1, msg2)
+	type args struct {
+		messages []messagebroker.ReceivedMessage
+	}
+	tests := []struct {
+		name     string
+		args     args
+		wantErr  bool
+		expected messagebroker.ReceivedMessage
+	}{
+		{
+			name:    "Test Delay Consumer one message counsume out of 2",
+			wantErr: false,
+			args: args{
+				messages: msgs,
+			},
+			expected: msg2,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			count := 0
+			consumer.EXPECT().ReceiveMessages(gomock.AssignableToTypeOf(subCtx), gomock.Any()).DoAndReturn(
+				func(arg0 context.Context, arg1 messagebroker.GetMessagesFromTopicRequest) (*messagebroker.GetMessagesFromTopicResponse, error) {
+					if count == 0 {
+						count++
+						return &messagebroker.GetMessagesFromTopicResponse{Messages: tt.args.messages}, nil
+					} else {
+						return &messagebroker.GetMessagesFromTopicResponse{}, nil
+					}
+				}).AnyTimes()
+			go dc.Run(ctx)
+			select {
+			case err := <-errCh:
+				if !tt.wantErr {
+					t.Errorf("Got Error : %v", err)
+				}
+			case <-time.NewTicker(time.Millisecond * 500).C:
+				assert.Equal(t, len(dc.cachedMsgs), 1)
+				assert.True(t, reflect.DeepEqual(dc.cachedMsgs[0], tt.expected))
+			}
+			cancel()
+			<-time.NewTicker(time.Millisecond * 1).C
+		})
+	}
+}
+
+func TestDelayConsumer_Run_DeadLetter(t *testing.T) {
+	ctx, ctrl, subCtx, cancel, brokerstore, handler, consumer, cache := setupDC(t)
+	subs := getDummySubscriptionModel()
+	subscriberID := "subscriber-dl-id"
+	errCh := make(chan error)
+	dlqTopicChan := make(chan *messagebroker.SendMessageToTopicRequest)
+	producer := buildMockDLQProducer(ctx, ctrl, dlqTopicChan)
+	brokerstore.EXPECT().GetProducer(gomock.AssignableToTypeOf(subCtx), messagebroker.ProducerClientOptions{
+		Topic:     "dlt1",
+		TimeoutMs: 100,
+	}).Return(producer, nil).AnyTimes()
+	dc, err := NewDelayConsumer(subCtx, subscriberID, "t1-dl", subs, brokerstore, handler, cache, errCh)
+	assert.NotNil(t, dc)
+	assert.Nil(t, err)
+	assert.NotNil(t, dc.LogFields())
+	tFut := time.Now().Add(time.Second * 10)
+	msgs := make([]messagebroker.ReceivedMessage, 0)
+	msg := getDummyBrokerMessageDLQ()
+	msg.NextDeliveryTime = tFut
+	msgs = append(msgs, msg)
+	count := 0
```

Comment:
can we add spaces b/w these, and comments, or split it into funcs. A lot of things are happening here,